test/widgets/message_list_test.dart

```diff
@@ -648,6 +645,31 @@ void main() {
       check(isMarkAsReadButtonVisible(tester)).isFalse();
     });
 
+    testWidgets('messages don\'t shift position', (WidgetTester tester) async {
+      final message = eg.streamMessage(flags: []);
+      final unreadMsgs = eg.unreadMsgs(streams:[
+        UnreadStreamSnapshot(topic: message.subject, streamId: message.streamId,
+          unreadMessageIds: [message.id])
+      ]);
+      await setupMessageListPage(tester,
+        messages: [message], unreadMsgs: unreadMsgs);
+      check(isMarkAsReadButtonVisible(tester)).isTrue();
+      check(tester.widgetList(find.byType(MessageItem)).length).equals(1);
```

Comment:
nit: do more processing on the result of `check`, instead of its argument:
```suggestion
      check(tester.widgetList(find.byType(MessageItem))).length.equals(1);
```

This allows `package:checks` to give more informative output if the check fails.

Reply:
Okay. Fixed.

Reply:
Okay. Fixed.